### PR TITLE
UIIN-2605: ECS: Check when sharing instance with source=MARC is complete before re-fetching it

### PR DESCRIPTION
**Pull request by mariia-aloshyna**
## Purpose

Sharing instance with source = MARC takes more time than instance with source = FOLIO. It needs to make sure that instance sharing process is complete before showing an updated instance to a user.

## Approach

- After a user hits the `Share local instance` action item -> send POST request to share the instance
- After that use setInterval to send GET request and check whether sharing is completed or not
- Once it's complete -> refetch the instance and render a shadow copy
- Move `isLoading` logic to `InstanceDetails` component

## Refs

https://issues.folio.org/browse/UIIN-2605

## Screenshots

https://github.com/folio-org/ui-inventory/assets/55138456/8c894120-04a2-414c-926c-75a7305f65dc

### CODE REVIEW

**Review by BogdanDenis — APPROVED** [review]

File: src/Instance/InstanceDetails/InstanceDetails.js [hunk 1]
```diff
@@ -57,24 +66,72 @@ const accordions = {
...
+        id="ui-inventory.instanceRecordSubtitle"
+        values={{
+          hrid: instance?.hrid,
+          updatedDate: getDate(instance?.metadata?.updatedDate),
+        }}
+      />
+    );
+  };
+
   const detailsLastMenu = useMemo(() => {
```
Comment (Dmytro-Melnyshyn) [inline src/Instance/InstanceDetails/InstanceDetails.js:135]: Please fix eslint warning `React Hook "useMemo" is called conditionally.`

File: src/ViewInstance.js [hunk 2]
```diff
@@ -414,6 +411,49 @@ class ViewInstance extends React.Component {
...
+
+          if (sharingStatus === INSTANCE_SHARING_STATUSES.ERROR) {
+            clearInterval(interval);
+            onError(response);
+          }
+        };
+
+        this.checkInstanceSharingProgress({ sourceTenantId, instanceIdentifier })
+          .then(onSuccess)
+          .catch(onError);
```
Comment (Dmytro-Melnyshyn) [inline src/ViewInstance.js:452]: `onError` is called in two places but `clearInterval` is invoked once, let's move `clearInterval` to `onError` cb.

Comment (mariia-aloshyna) [reply src/ViewInstance.js]: done

File: src/Instance/InstanceDetails/InstanceDetails.js (hunk 1 above)
Comment (mariia-aloshyna) [reply src/Instance/InstanceDetails/InstanceDetails.js]: done

File: src/ViewInstance.js [hunk 3]
```diff
@@ -414,6 +411,48 @@ class ViewInstance extends React.Component {
...
+            message: <FormattedMessage id="ui-inventory.shareLocalInstance.toast.unsuccessful" values={{ instanceTitle }} />,
+          });
+          clearInterval(interval);
+          reject(error);
+        };
+        const onSuccess = response => {
+          const sharingStatus = response?.sharingInstances[0]?.status;
+
+          if (sharingStatus === INSTANCE_SHARING_STATUSES.COMPLETE) {
+            clearInterval(interval);
```
[the remark this answers is not in the document]
Comment (mariia-aloshyna) [reply src/ViewInstance.js:442]: good point, adjusted

File: src/ViewInstance.js [hunk 4]
```diff
@@ -450,6 +452,8 @@ class ViewInstance extends React.Component {
           .then(onSuccess)
           .catch(onError);
       }, 2000);
+
+      this.setState({ intervalId: interval });
```
Comment (Dmytro-Melnyshyn) [inline src/ViewInstance.js:456]: There is no need to use state, we don't want to change anything on the UI, so let's use `this.intanceSharingInterval = setInterval(...)`

Comment (mariia-aloshyna) [reply src/ViewInstance.js]: what about now?

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Pull request merged**